### ISSUE TITLE
Rebuild of assets will occasionally not happen

### ISSUE THREAD

**Issue by AndrewIsh** [jhthorsen/mojolicious-plugin-assetpack#120]
I've no way of reliably replicating this, just noting it in case anyone else can. I've noticed that, on occasions, AssetPack will not rebuild changed assets.

**Comment by jhthorsen** [comment]
I've experienced the same issue myself, but I don't know how to reproduce it.

The fix for now is to delete `assets/assets/cache/topic-name.{js,css}` which forces AssetPack to rebuild the asset.

If someone has additional information about how to reproduce the problem, then please explain the steps and/or submit a failing test.

**Comment by AndrewIsh** [comment]
Just to add to this as I'm still experiencing it on a several times a day basis. I've noticed that it will often happen on two consecutive occassions. So, restart, rebuild fails, restart again, rebuild fails, restart again, rebuild succeeds.
No idea if that's in any way helpful, but thought it worth mentioning.

**Comment by jhthorsen** [comment]
Could you try to run the code below, before loading AssetPack plugin? I'm not sure if this is the real bug, but it would be nice to see if it helps.

```
use Mojolicious::Plugin::AssetPack::Util;
$Mojolicious::Plugin::AssetPack::Util::SUM_LEN = 16;
```

https://github.com/jhthorsen/mojolicious-plugin-assetpack/blob/master/lib/Mojolicious/Plugin/AssetPack/Util.pm#L14

**Comment by jhthorsen** [comment]
I think I've figured it out, but I don't get why it doesn't happen more often if I'm right: AssetPack calculates the output checksum based on the input files. This means that if you track assetpack.db in git, while not tracking the output files, then the checksums will match and AssetPack will skip processing the asset.

Does this sound familiar? Do you track all files in git?

I've tried to update the documentation https://metacpan.org/pod/distribution/Mojolicious-Plugin-AssetPack/lib/Mojolicious/Plugin/AssetPack/Guides/Developing.pod#Faster-development-cycle, but forgot to mention that MOJO_ASSETPACK_LAZY will also skip tracking the checksums for processed assets.

**Comment by jhthorsen** [comment]
I think this will be fixed by #137.
